honeybee_energy/schedule/fixedinterval.py

```diff
@@ -0,0 +1,690 @@
+                DataCollection. Default: 31 Dec on a non-leap year.
+        """
+        a_period = AnalysisPeriod(start_date.month, start_date.day, 0,
+                                  end_date.month, end_date.day, 23, timestep,
+                                  self.is_leap_year)
+        data_type, unit = self._get_lb_data_type_and_unit()
+        header = Header(data_type, unit, a_period, metadata={'schedule': self.name})
+        values = self.values_at_timestep(timestep, start_date, end_date)
+        return HourlyContinuousCollection(header, values)
+
+    @classmethod
+    def from_idf(cls, idf_string, type_idf_string=None):
+        """Create a ScheduleFixedInterval from an EnergyPlus IDF text strings.
+
+        Args:
+            idf_string: A text string fully describing an EnergyPlus
+                Schedule:File.
+            type_idf_string: An optional text string for the ScheduleTypeLimits.
+                If None, the resulting schedule will have no ScheduleTypeLimit.
+        """
+        # process the schedule inputs
+        sch_fields = parse_idf_string(idf_string, 'Schedule:File')
+        schedule_type = ScheduleTypeLimit.from_idf(type_idf_string) if type_idf_string \
+            is not None else None
+        timestep = 60 / int(sch_fields[8]) if sch_fields[8] != '' else 1
+        start_date = Date(1, 1, False) if sch_fields[5] == '8760' else Date(1, 1, True)
+        interpolate = False if sch_fields[7] == 'No' or sch_fields[7] == '' else True
+
+        # load the data from the CSV file referenced in the string
+        assert os.path.isfile(sch_fields[2]), \
+            'CSV Schedule:File "{}" was not found on this system.'.format(sch_fields[2])
+        all_data = csv_to_matrix(sch_fields[2])
+        transposed_data = tuple(zip(*all_data))
+        csv_data = (float(x) for x in
+                    transposed_data[int(sch_fields[3]) - 1][int(sch_fields[4]):])
+
+        return cls(sch_fields[0], csv_data, schedule_type, timestep, start_date,
+                   0, interpolate)
+
+    @classmethod
+    def from_dict(cls, data):
+        """Create a ScheduleFixedInterval from a dictionary.
+
+        Note that the dictionary must be a non-abridged version for this
+        classmethod to work.
+
+        Args:
+            data: ScheduleFixedInterval dictionary following the format below.
+
+        .. code-block:: shell
+
+            {
+            "type": 'ScheduleFixedInterval',
+            "name": 'Automated Awning Transmittance',
+            "values": [], // list of numbers for the values of the schedule
+            "schedule_type_limit": {}, // ScheduleTypeLimit dictionary representation
+            "timestep": 1, // Integer for the timestep of the schedule
+            "start_date": {}, // Date dictionary representation
+            "out_of_range_value": 0, // Number for the values out of range
+            "interpolate": False // Boolean noting whether to interpolate between values
+            }
```

Comment:
we will see once this is merged but I think you need an extra empty line after the end of the code-block.

Reply:
Ah, Ok. I added the extra line break.

Reply:
Actaully, I just checked my old code where I had done this and the extra line break is unnecessary:
https://www.ladybug.tools/honeybee-energy/docs/honeybee_energy.schedule.day.html